components/brave_adaptive_captcha/brave_adaptive_captcha_service.cc

```diff
@@ -0,0 +1,189 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_adaptive_captcha/brave_adaptive_captcha_service.h"
+
+#include <algorithm>
+#include <utility>
+
+#include "base/strings/stringprintf.h"
+#include "brave/components/api_request_helper/api_request_helper.h"
+#include "brave/components/brave_adaptive_captcha/environment.h"
+#include "brave/components/brave_adaptive_captcha/server_util.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+namespace brave_adaptive_captcha {
+
+constexpr int kScheduledCaptchaMaxFailedAttempts = 10;
+
+Environment BraveAdaptiveCaptchaService::environment_ = DEVELOPMENT;
+
+const char kScheduledCaptchaId[] = "brave.rewards.scheduled_captcha.id";
+const char kScheduledCaptchaPaymentId[] =
+    "brave.rewards.scheduled_captcha.payment_id";
+const char kScheduledCaptchaSnoozeCount[] =
+    "brave.rewards.scheduled_captcha.snooze_count";
+const char kScheduledCaptchaFailedAttempts[] =
+    "brave.rewards.scheduled_captcha.failed_attempts";
+const char kScheduledCaptchaPaused[] = "brave.rewards.scheduled_captcha.paused";
+
+net::NetworkTrafficAnnotationTag kAnnotationTag =
+    net::DefineNetworkTrafficAnnotation("brave_adaptive_captcha_service", R"(
+        semantics {
+          sender:
+            "Brave Adaptive Captcha service"
+          description:
+            "Fetches CAPTCHA data from Brave."
+          trigger:
+            "The Brave service indicates that it's time to solve a CAPTCHA."
+          data: "Brave CAPTCHA data."
+          destination: WEBSITE
+        }
+        policy {
+          cookies_allowed: NO
+          setting:
+            "This feature cannot be disabled by settings."
+          policy_exception_justification:
+            "Not implemented."
+        })");
+
+BraveAdaptiveCaptchaService::BraveAdaptiveCaptchaService(
+    content::BrowserContext* context,
```

Comment:
this is unused and creates a dependency on content that we don't want here

Reply:
Fixed.